src/main/kotlin/no/fdk/dataservicecatalog/handler/DataServiceHandler.kt

```diff
@@ -74,6 +76,28 @@ class DataServiceHandler(private val repository: DataServiceRepository) {
 
         logger.info("Deleted Data Service with id: $dataServiceId in Catalog with id: $catalogId")
     }
+
+    fun publish(catalogId: String, dataServiceId: String) {
+        val dataService = (repository.findDataServiceById(dataServiceId)
+            ?.takeIf { it.catalogId == catalogId }
+            ?.also { if (it.published) throw BadRequestException("Data Service with id: $dataServiceId is already published") }
```

Comment:
Kan vurdere å ikkje bruke also til å kaste exceptions. Er mest ment for side-effekt frie ting som logging og debugging.

Reply:
👍 
7fa13e861a29eb6f9efa1d5c7d6f5bddac4346a5